test/lsp/lsp_hover.ml

```diff
@@ -1141,3 +1141,116 @@ let%expect_test "hover-typedef-communication-section" =
     Hovering nothing worthy
     (line 10, character 17):
     Hovering nothing worthy |}];;
+
+let%expect_test "hover-comment" =
+  let { projdir; end_with_postproc }, server = make_lsp_project () in
+  print_hovered server ~projdir @@ extract_position_markers {cobol|
+>>source format is fixed
+       IDENTIFICATION DIVISION.
+       PROGRAM-ID. prog.
+       DATA DIVISION.
+       WORKING-STORAGE SECTION.
+      * full line comment
+       01 STRUCT. *> inline comment
+         02 VAL-1 PIC X. *> val1 inline comment
+         02 VAL-2 PIC X.
+       PROCEDURE DIVISION.
+         DISPLAY S_|_TRUCT V_|_AL-1 V_|_AL-2.
+         STOP RUN.
+    |cobol};
+  end_with_postproc [%expect.output];
+  [%expect {|
+    {"params":{"diagnostics":[],"uri":"file://__rootdir__/prog.cob"},"method":"textDocument/publishDiagnostics","jsonrpc":"2.0"}
+    (line 11, character 18):
+    __rootdir__/prog.cob:12.17-12.23:
+       9            02 VAL-1 PIC X. *> val1 inline comment
+      10            02 VAL-2 PIC X.
+      11          PROCEDURE DIVISION.
+      12 >          DISPLAY STRUCT VAL-1 VAL-2.
+    ----                    ^^^^^^
+      13            STOP RUN.
+      14
+    ```cobol
+    STRUCT
+    ```
+    Group of 2 subfields
+    Size: 16 bits
+    ```cobol
+    * full line comment
+    *> inline comment
```

Comment:
I don't think the two kinds of comments should be shown. In this case I think only the inline comment should be considered. Plus not inside a COBOL code block either, and without the comment markers (juste assuming it's plain text — that could be checked later via some heuristics).